File: Math/Problem_9.c
//Palindrome Number

#include <stdbool.h>

bool isPalindrome(int x) {
    if (x < 0) {
        return false;
    } else {
        int orginal = x, remainder;
        long long reversed = 0;

        while (x != 0) {
            remainder = x % 10;
            reversed = reversed * 10 + remainder;
            x /= 10;
        }

        if (orginal == reversed) {
            return true;
        } else {
            return false;
        }
    }
}

File: Go/Problem_9.go
package main

import (
	"fmt"
)

func isPalindrome(value int) bool {
	if value < 0 {
		return false
	}

	var original int = value
	var reversed int
	var remainder int = 0

	for value > 0 {
		remainder = value % 10
		reversed = reversed*10 + remainder
		value /= 10
	}

	if original == reversed {
		return true
	} else {
		return false
	}

}

func execute_problem9(palindrome int) {

	var buffer bool = isPalindrome(palindrome)

	if buffer == true {
		fmt.Printf("Palindrome\n")
	} else {
		fmt.Printf("Not Palindrome\n")
	}

}
